arkouda/dataframe.py

```diff
@@ -5326,6 +5609,322 @@ def from_return_msg(cls, rep_msg):
         return cls(columns, idx)
 
 
+class _LocIndexer:
+    def __init__(self, df):
+        self.df = df
+
+    def __getitem__(self, key):
+        if isinstance(key, tuple) and len(key) == 2:
+            return self._get_row_col(key[0], key[1])
+        if isinstance(key, list):
+            key = array(key)
+        if isinstance(key, Series):
+            key = key.values
+        if is_supported_scalar(key) and self.df.index.dtype == dtype(type(key)):
+            return self.df._get_rows(indexof1d(array([key]), self.df.index.values))
+
+        if isinstance(key, pdarray) and key.dtype == self.df.index.dtype:
+            return self.df._get_rows(indexof1d(key, self.df.index.values))
+
+        if isinstance(key, slice):
+            if key.start is not None and not akany(in1d(array([key.start]), self.df.index.values)):
+                raise KeyError(f"Index {key.start} not found in DataFrame index")
+            if key.stop is not None and not akany(in1d(array([key.stop]), self.df.index.values)):
+                raise KeyError(f"Index {key.stop} not found in DataFrame index")
+
+            start_idx = (indexof1d(array([key.start]), self.df.index.values)[0]
+                         if key.start is not None else 0)
+            stop_idx = (indexof1d(array([key.stop]), self.df.index.values)[0] + 1
+                        if key.stop is not None else self.df.index.size)
```

Comment:
okay im sorry if im starting to sound like a broken record lol, but if we only want the first occurrence (which it seems we do since we're indexing at `0`, `find` is the way to go.

```python
start_idx = (find(array([key.start]), self.df.index.values) if key.start is not None else 0)
stop_idx = (find(array([key.stop]), self.df.index.values) + 1 if key.stop is not None else self.df.index.size)
```

Reply:
you are correct, we want the first occurrence.